File: createJson/src/main/java/com/ruzhan/jsonfile/CreateJsonMain.java
package com.ruzhan.jsonfile;

import com.google.gson.Gson;
import com.ruzhan.jsonfile.helper.MovieHelper;
import com.ruzhan.jsonfile.model.Movie;
import com.ruzhan.jsonfile.model.MovieDetail;

import java.util.List;
import java.util.Map;

public class CreateJsonMain {

    public static final String FILE_TYPE = ".json";
    private static final String MOVIE_LIST = "E:\\project\\Lion\\json\\api\\movie";
    private static final String MOVIE_DETAIL = "E:\\project\\Lion\\json\\api\\movie\\detail";

    public static final int CODE_SUCCESS = 200;

    private static Gson gson = new Gson();

    public static void main(String[] args) {
        System.out.println("init json file...");
        System.out.println("=================================");

        System.out.println("1, create movie list json files");
        List<Movie> movieList = MovieHelper.getMovieList();
        Map<String, List<Movie>> movieMap = MovieHelper.getMovieMap(movieList);
        System.out.println("movieMap size:" + movieMap.size());
        MovieHelper.movieListToJsonFile(movieMap, MOVIE_LIST, gson);
        System.out.println("1, create movie list json files finish ... !!!");
        System.out.println("=================================");

        System.out.println("2, create movie detail json files");
        List<MovieDetail> detailList = MovieHelper.getMovieDetailList();
        MovieHelper.movieDetailListToJsonFile(detailList, MOVIE_DETAIL, gson);
        System.out.println("2, create movie detail json files finish ... !!!");
        System.out.println("=================================");


        System.out.println("json file create finish ... !!!");
    }
}

File: createJson/src/main/java/com/ruzhan/jsonfile/CreateJsonMain.kt
package com.ruzhan.jsonfile

import com.google.gson.Gson
import com.ruzhan.jsonfile.helper.MovieHelper
import java.io.File

object CreateJsonMain {

    private const val USER_DIR = "user.dir"
    private const val JSON = "json"
    private const val API = "api"
    private const val MOVIE = "movie"
    private const val DETAIL = "detail"
    const val FILE_TYPE = ".json"

    private val mainGSon = Gson()
    private var movieListFile = File("")
    private var movieDetailFile = File("")

    @JvmStatic
    fun main(args: Array<String>) {
        initCreateJsonFile()
        createMovieListToJsonFile()
        createMovieDetailListToJsonFile()
        println("=======")
        println("==============")
        println("======= CreateJsonMain finish !!! =======")
    }

    private fun initCreateJsonFile() {
        println("initCreateJsonFile call...")
        val userDirFile = File(System.getProperty(USER_DIR))
        println("userDirFile:${userDirFile.absolutePath}")

        val createJsonFile = File(File(userDirFile, JSON), API)
        println("createJsonFile:${createJsonFile.absolutePath}")

        movieListFile = File(createJsonFile, MOVIE)
        println("movieListFile:${movieListFile.absolutePath}")

        movieDetailFile = File(movieListFile, DETAIL)
        println("movieDetailFile:${movieDetailFile.absolutePath}")
        println("======= initCreateJsonFile finish !!! =========")
    }

    private fun createMovieListToJsonFile() {
        println("createMovieListToJsonFile call...")
        val movieList = MovieHelper.movieList
        val movieMap = MovieHelper.getMovieMap(movieList)
        println("createMovieListToJsonFile movieMap size:" + movieMap.size)
        MovieHelper.movieListToJsonFile(movieMap, movieListFile.absolutePath, mainGSon)
        println("======== createMovieListToJsonFile finish !!! ==========")
    }

    private fun createMovieDetailListToJsonFile() {
        println("createMovieDetailListToJsonFile call...")
        val detailList = MovieHelper.movieDetailList
        MovieHelper.movieDetailListToJsonFile(detailList, movieDetailFile.absolutePath, mainGSon)
        println("======= createMovieDetailListToJsonFile finish !!! ==========")
    }
}
